### PR TITLE
chore: merge property files instead of override and fix flaky tests

### PR DESCRIPTION
**Pull request by manolo**

### CODE REVIEW

File: scripts/mergeITs.js [hunk 1]
```diff
@@ -111,13 +111,21 @@ async function createPom() {
...
-  //if target is a directory a new file with the same name will be created
+  // if target is a directory a new file with the same name will be created
   if (fs.existsSync(target)) {
     if (fs.lstatSync(target).isDirectory()) {
       targetFile = path.join(target, path.basename(source));
     }
   }
   if (fs.existsSync(targetFile)) {
+    // When file exists we can merge both or override depending on the type
+    if (/.properties$/.test(source)) {
```
[the remark this answers is not in the document]
Comment (manolo) [reply scripts/mergeITs.js:122]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply scripts/mergeITs.js:122]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply scripts/mergeITs.js:122]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply scripts/mergeITs.js:122]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply scripts/mergeITs.js:122]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply scripts/mergeITs.js:122]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply scripts/mergeITs.js:122]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (manolo) [reply scripts/mergeITs.js:122]: @sonatype-lift ignore

File: scripts/mergeITs.js [hunk 2]
```diff
@@ -111,13 +111,21 @@ async function createPom() {
...
+      let content = fs.readFileSync(source, 'utf8');
+      content += fs.readFileSync(targetFile, 'utf8');
```
Comment (sissbruecker) [inline scripts/mergeITs.js:124]: This kind of relies on having a trailing newline in the first file, which isn't enforced. Maybe we should add a newline between both file contents?

File: vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogHeaderFooterIT.java [hunk 3]
```diff
@@ -211,7 +203,25 @@ private void assertDialogNotContains(String text) {
...
     private void verifyDialogOpened() {
-        waitForElementPresent(By.tagName("vaadin-dialog-overlay"));
+        waitForElementPresent(
+                By.cssSelector("vaadin-dialog-overlay div button"));
+    }
+
+    private void verifyContentRendered(String content) {
+        waitUntil(c -> {
+            try {
+                for (;;) {
```
Comment (sissbruecker) [inline vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogHeaderFooterIT.java:213]: I don't think this loop is necessary, `waitUntil` already repeats the code below in an interval.

File: vaadin-iron-list-flow-parent/vaadin-iron-list-flow-integration-tests/src/test/java/com/vaadin/flow/component/ironlist/tests/IronListViewIT.java [hunk 4]
```diff
@@ -123,7 +123,7 @@ public void disabledListWithTemplates() throws InterruptedException {
 
     @Test
     public void rankedListWithEventHandling() {
-        validateListSize(findElement(By.id("using-events-with-templates")), 29);
+        validateListSize(findElement(By.id("using-events-with-templates")), 20);
```
Comment (sissbruecker) [inline vaadin-iron-list-flow-parent/vaadin-iron-list-flow-integration-tests/src/test/java/com/vaadin/flow/component/ironlist/tests/IronListViewIT.java:126]: I guess this effectively disables virtualization, which might be missing the point for a virtual list component. I'd suggest to disable the test instead and open a ticket to properly fix it.

File: vaadin-virtual-list-flow-parent/vaadin-virtual-list-flow-integration-tests/src/test/java/com/vaadin/flow/component/virtuallist/tests/VirtualListViewIT.java [hunk 5]
```diff
@@ -121,7 +121,7 @@ public void disabledListWithTemplates() throws InterruptedException {
 
     @Test
     public void rankedListWithEventHandling() {
-        validateListSize(findElement(By.id("using-events-with-templates")), 29);
+        validateListSize(findElement(By.id("using-events-with-templates")), 20);
```
Comment (sissbruecker) [inline vaadin-virtual-list-flow-parent/vaadin-virtual-list-flow-integration-tests/src/test/java/com/vaadin/flow/component/virtuallist/tests/VirtualListViewIT.java:124]: I guess this effectively disables virtualization, which might be missing the point for a virtual list component. I'd suggest to disable the test instead and open a ticket to properly fix it.

File: scripts/mergeITs.js (hunk 2 above)
Comment (manolo) [reply scripts/mergeITs.js]: done

File: vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogHeaderFooterIT.java (hunk 3 above)
Comment (manolo) [reply vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogHeaderFooterIT.java]: done

File: vaadin-virtual-list-flow-parent/vaadin-virtual-list-flow-integration-tests/src/test/java/com/vaadin/flow/component/virtuallist/tests/VirtualListViewIT.java (hunk 5 above)
Comment (manolo) [reply vaadin-virtual-list-flow-parent/vaadin-virtual-list-flow-integration-tests/src/test/java/com/vaadin/flow/component/virtuallist/tests/VirtualListViewIT.java:124]: changed generator for a while loop so as there are always 20

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**